### DISCUSSION TITLE
upper/lower case in domain table (can not add new mailbox)

### DISCUSSION THREAD

**Discussion by shpatel579** [postfixadmin/postfixadmin#741] [Q&A]
Hello,
after a series of postfixadmin upgrades (to 3.1 then to 3.3.9), I have noticed that I cannot add new mailboxes. There is no error message, just a red rectangle with a single dot at the last step of the wizard.
With the postfixadmin-cli 'mailbox add' I got 'Error: Invalid value given for domain' message, which is not true, since 'domain view' shows it.
After short investigation, I have found out that issue probably related to the mixture of upper/lower case in the mysql database domain table.

As a temporary workaround, I have patched the [ 'functions.inc.php'](https://github.com/postfixadmin/postfixadmin/blob/71f17de6b170698f94bbf0dd2cdd0a9f0079eff1/functions.inc.php) file as follows: [function.inc.php.patch.txt](https://github.com/postfixadmin/postfixadmin/files/11939702/function.inc.php.patch.txt)

Thus, I managed to pass the mailbox creation wizard successfully. However, there are some other errors, so I suppose this is not a proper solution.

The domain was created more than 10 years ago, so it was some ancient version of postfixadmin.
How can I resolve the problem in a safe and proper way?

**Comment by DavidGoodwin** [comment]
Hm, MySQL normally uses a case insensitive collation - so I wouldn't expect this to be a problem ... unless you have a non-ci collation set?

try doing a `show create table mailbox`  when logged in to MySQL?

**Reply by shpatel579** [reply]
Please, see below:
```
CREATE TABLE `mailbox` (
  `username` varchar(255) NOT NULL,
  `password` varchar(255) NOT NULL,
  `name` varchar(255) CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL,
  `maildir` varchar(255) NOT NULL,
  `quota` bigint(20) NOT NULL DEFAULT '0',
  `local_part` varchar(255) NOT NULL,
  `domain` varchar(255) NOT NULL,
  `created` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `modified` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `active` tinyint(1) NOT NULL DEFAULT '1',
  `phone` varchar(30) CHARACTER SET utf8 NOT NULL DEFAULT '',
  `email_other` varchar(255) CHARACTER SET utf8 NOT NULL DEFAULT '',
  `token` varchar(255) CHARACTER SET utf8 NOT NULL DEFAULT '',
  `token_validity` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `password_expiry` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  PRIMARY KEY (`username`),
  KEY `domain` (`domain`)
) ENGINE=MyISAM DEFAULT CHARSET=latin1 COMMENT='Postfix Admin - Virtual Mailboxes'
```

**Comment by DavidGoodwin** [comment]
I think you're suffering from not having the latest schema updates applied.

See also #327

**Reply by shpatel579** [reply]
I tried to re-run setup.php and it shows: 
> Database Update
> Everything seems fine... attempting to create/update database structure
> Database is up to date: 1844/0

But it looks like nothing changed in the database.
Should I upgrade to the newer version of the postfixadmin (now at v3.3.9)?

**Reply by DavidGoodwin** [reply]
well, the latest stable release is https://github.com/postfixadmin/postfixadmin/releases/tag/postfixadmin-3.3.15 ...
